File: tests/nodejs/cases/case_034_archive_shipment.js
/**
 * Task case 034: archive_shipment
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case034ArchiveShipment(input) {
  // Task: archive shipment records and prepare transport-ready payload.
  const taskName = 'archive_shipment';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'shipment',
    stage: 'archive',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  // ESCAPE: payload is nested inside retained envelope persisted to retained audit state.
  const envelope = { kind: 'audit-envelope', payload };
  retainedAudit.push(envelope);
  return 'ok';
}

module.exports = {
  case034ArchiveShipment,
};

File: tests/go/cases/case_034_archive_shipment.go
package escape_tests

var retainedCase034 = []map[string]string{}

func Case034ArchiveShipment(input string) string {
    // Task case 034: archive shipment payload for downstream business processing.
    raw := input
    if raw == "" {
        raw = "sample"
    }
    payload := map[string]string{
        "task": "archive_shipment",
        "entity": "shipment",
        "stage": "archive",
        "input": raw,
    }
    // ESCAPE: payload is wrapped in retained envelope for downstream replay.
    envelope := map[string]string{"source": "pipeline", "payload": payload["task"]}
    retainedCase034 = append(retainedCase034, envelope)
    return "ok"
}
